micro-rdk/src/common/board.rs

```diff
@@ -199,8 +197,12 @@ impl Board for FakeBoard {
     fn get_analog_reader_by_name(
         &self,
         name: String,
-    ) -> Result<Rc<RefCell<dyn AnalogReader<u16, Error = AnalogError>>>, BoardError> {
-        match self.analogs.iter().find(|a| a.borrow().name() == name) {
+    ) -> Result<Arc<Mutex<dyn AnalogReader<u16, Error = AnalogError>>>, BoardError> {
+        match self
+            .analogs
+            .iter()
+            .find(|a| a.lock().unwrap().name() == name)
```

Comment:
[Why] do you still need the explicit `lock().unwrap()` here?

Reply:
hmm, I probably don't? I don't remember why I added this